codec-http/src/main/java/io/netty/handler/codec/http/multipart/DefaultHttpDataFactory.java

```diff
@@ -241,7 +246,16 @@ public FileUpload createFileUpload(HttpRequest request, String name, String file
     public void removeHttpDataFromClean(HttpRequest request, InterfaceHttpData data) {
         if (data instanceof HttpData) {
             List<HttpData> fileToDelete = getList(request);
-            fileToDelete.remove(data);
+
+            // Can't simply call fileToDelete.remove(data), need to check identity
```

Comment:
can you explain why ?

Reply:
Thanks, I've improved the comment to explain why.